src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -565,10 +565,17 @@ public int Length
         ///      foreach (ReadOnlyMemory<char> chunk in sb.EnumerateChunks())
         ///         foreach(char c in chunk.Span)
         ///             { /* operation on c }
-        ///      
-        /// Note that creating a ReadOnlySpan from a ReadOnlyMemory is expensive compared to the
-        /// fetching of the character, so create a local variable for the SPAN if you need to use
-        /// a for statement for example 
+        ///
+        /// It is undefined what happens if the StringBuilder is modified while the chunk
+        /// enumeration is incomplete.  StringBuilder is also not thread-safe, so operating
+        /// on it with concurrent threads is illegal.  Finally the ReadOnlyMemory chunks returned 
```

Comment:
Does this mean we'd consider it erroneous to have two threads both enumerating a StringBuilder's chunks concurrently (reading, not writing)?

Reply:
The statement is that the StringBuilder code is not thread-safe (this is not new, it has always been true).    In general non-tread-safe read-only code will  actually work if two threads execute it simultaneously, and that applies here as well, but I would not advertise that.  

If you want the wording tweeked, I am happy to change it, my main concern was to highlight that modification while enumeration (concurrent or not), is bad.

Reply:
It's fine.  I just wanted to make sure I understood, e.g. you could imagine that EnumerateChunks could actually mutate the StringBuilder, in which case concurrent reading would be a bad idea.